app/controllers/api/events_controller.rb

```diff
@@ -0,0 +1,41 @@
+class API::EventsController < ApplicationController
+
+  skip_before_action  :verify_authenticity_token
+  before_filter :set_access_control_headers
+
+  def create
+
+    @registered_app = RegisteredApp.find_by(url: request.env['HTTP_ORIGIN'])
+    @event = Event.new(event_params)
+    @event.registered_app_id = @registered_app.id
+
+    if @registered_app == nil
+
+      render json: "Sorry. This is an unregistered application", status: :unprocessable_entity
+    elsif @event.valid?
+      @event.save!
```

Comment:
what happens if this `save!` fails for some reason?  What if you used `@event.save` in place of `@event.valid?` ?